apis/datadoghq/v1alpha1/datadogagent_types.go

```diff
@@ -947,11 +947,16 @@ type CRISocketConfig struct {
 // DogstatsdConfig contains the Dogstatsd configuration parameters.
 // +k8s:openapi-gen=true
 type DogstatsdConfig struct {
-	// Enable origin detection for container tagging.
+	// Enable origin detection for container tagging with UDS.
 	// See also: https://docs.datadoghq.com/developers/dogstatsd/unix_socket/#using-origin-detection-for-container-tagging
 	// +optional
 	DogstatsdOriginDetection *bool `json:"dogstatsdOriginDetection,omitempty"`
 
+	// Enable origin detection for container tagging with UDP.
+	// See also: https://docs.datadoghq.com/developers/dogstatsd/?tab=hostagent#how-it-works
+	// +optional
+	DogstatsdOriginDetectionClient *bool `json:"dogstatsdOriginDetectionClient,omitempty"`
```

Comment:
I think we should have a single config option for both origin detection and origin detection client similarly to https://github.com/DataDog/helm-charts/pull/1297

Reply:
I've changed the code completely 👍